src/store/reducers/movieReducer.ts

```diff
@@ -0,0 +1,92 @@
+import { MOVIES_TYPES } from "../types";
+import axios from "axios";
+import { Movie, MoviesResponse, ReduxAction } from "../../components/MovieCard/interface";
+
+const initialState = {
+    movies: [],
+    total: 0,
+    selectedGenres: [],
+    selectedSorting: "release_date"
+}
+
+export default function movieReducer(state: typeof initialState = initialState, action: ReduxAction) {
+    switch (action.type) {
+        case MOVIES_TYPES.LOAD_MOVIES: {
+            return {
+                ...state,
+                movies: [
+                    ...action.payload
+                ]
+            }
+        }
+        case MOVIES_TYPES.UPDATE_MOVIES_COUNT: {
+            return {
+                ...state,
+                total: action.payload
+            }
+        }
+        case MOVIES_TYPES.UPDATE_SELECTED_GENRES: {
+            return {
+                ...state,
+                selectedGenres: [
+                    ...action.payload
+                ]
+            }
+        }
+        case MOVIES_TYPES.UPDATE_SELECTED_SORTING: {
+            return {
+                ...state,
+                selectedSorting: action.payload
+            }
+        }
+        default:
+            return state
+    }
+}
+
+export const updateMovies = (movies: Movie[]) => ({
+    type: MOVIES_TYPES.LOAD_MOVIES,
+    payload: movies
+})
+
+export const updateMoviesCount = (count: number) => ({
+    type: MOVIES_TYPES.UPDATE_MOVIES_COUNT,
+    payload: count
+})
+
+export const updateSelectedGenres = (genres: string[]) => ({
+    type: MOVIES_TYPES.UPDATE_SELECTED_GENRES,
+    payload: genres
+})
+
+export const updateSelectedSorting = (title: string) => ({
+    type: MOVIES_TYPES.UPDATE_SELECTED_SORTING,
+    payload: title
+})
+
+export const fetchMovies = (limit: number = 9) => async (dispatch: any, getState: any) => {
```

Comment:
This one doesn't suit quite well into reducers module, because, it is not a [reducer](https://redux.js.org/tutorials/fundamentals/part-3-state-actions-reducers#writing-reducers) in the first place, furthermore, [you're supposed](https://redux.js.org/style-guide/style-guide#reducers-must-not-have-side-effects) to perform necessary side-effects using [specific middleware](https://redux.js.org/style-guide/style-guide#use-thunks-for-async-logic).

To sort this out, I would 
* come up with a separate module with [API endpoint](https://axios-http.com/docs/instance) and necessary CRUD-methods defined
* put together a separate module with necessary side-effects (preferably, based on Redux Saga)
* trigger necessary side-effect in response to specific event through dispatching appropriate action (which is caught by corresponding [middleware](https://redux.js.org/understanding/history-and-design/middleware))